src/channel_state.ts

```diff
@@ -424,6 +437,10 @@ export class ChannelState<
     }
   }
 
+  setIsUptoDate = (isUpToDate: boolean) => {
```

Comment:
does it make sense to move this to the channel level?

Reply:
Its only about messages, so I thought state would be a better place!!